src/frontend/src/expr/type_inference/func.rs

```diff
@@ -327,6 +327,25 @@ fn infer_type_for_special(
             .map(Some)
             .map_err(Into::into)
         }
+        ExprType::ConstantLookup => {
+            let len = inputs.len();
+            align_types(inputs.iter_mut().enumerate().filter_map(|(i, e)| {
+                // This optimized `ConstantLookup` organize `inputs` as
+                // [operand_const_expr]? (cond, res) [else]? pairs.
+                // So we align exprs at even indices as well as the last one
+                // when length is odd.
+                // Note that currently we assume fallback expression must exist
+                match (len.is_even() && i != 0 && i.is_even())
+                    || (len.is_odd() && i.is_odd())
+                    || i == len - 1
```

Comment:
It seems we can just use `(i != 0 && i.is_even()) || i == len - 1 `

Reply:
Indeed!